index.html

```diff
@@ -204,55 +204,1951 @@ <h3>
+          <h4>
+            <dfn>HIDDeviceRequestOptions</dfn> dictionary
+          </h4>
+          <pre class="idl">
+            dictionary HIDDeviceRequestOptions {
+                required sequence&lt;HIDDeviceFilter&gt; filters;
+            };
+          </pre>
+          <dl>
+            <dt>
+              <dfn>filters</dfn> member
+            </dt>
+            <dd>
+              Filters for HID devices.
+            </dd>
+          </dl>
+        </section>
+        <section data-dfn-for="HIDDeviceFilter">
+          <h4>
+            <dfn>HIDDeviceFilter</dfn> dictionary
+          </h4>
+          <pre class="idl">
+            dictionary HIDDeviceFilter {
+                unsigned long vendorId;
+                unsigned short productId;
+                unsigned short usagePage;
+                unsigned short usage;
+            };
+          </pre>
+          <dl>
+            <dt>
+              <dfn>vendorId</dfn> member
+            </dt>
+            <dd>
+              The vendor ID to match.
+            </dd>
+            <dt>
+              <dfn>productId</dfn> member
+            </dt>
+            <dd>
+              The product ID to match.
+            </dd>
+            <dt>
+              <dfn>usagePage</dfn> member
+            </dt>
+            <dd>
+              The [=top-level collection=] [=Usage Page=] to match.
+            </dd>
+            <dt>
+              <dfn>usage</dfn> member
+            </dt>
+            <dd>
+              The [=top-level collection=] [=Usage ID=] to match.
+            </dd>
+          </dl>
+          <p>
+            A {{HIDDevice}} |device:HIDDevice| <dfn data-lt=
+            "match the filter">matches the filter</dfn>
+            |filter:HIDDeviceFilter| if these steps return `true`:
+          </p>
+          <ol>
+            <li>If |filter|["{{HIDDeviceFilter/vendorId}}"] is present:
+              <ol>
+                <li>If |filter| contains "{{HIDDeviceFilter/productId}}" and
+                |device|["{{HIDDevice/productId}}"] is not equal to
+                |filter|["{{HIDDeviceFilter/productId}}"], return `false`.
+                </li>
+                <li>If |device|["{{HIDDevice/vendorId}}"] is not equal to
+                |filter|["{{HIDDeviceFilter/vendorId}}"], return `false`.
+                </li>
+              </ol>
+            </li>
+            <li>If |filter|["{{HIDDeviceFilter/usagePage}}"] is present:
+              <ol>
+                <li>Repeat for each {{HIDCollectionInfo}}
+                |collection:HIDCollectionInfo| of
+                |device|.{{HIDDevice/collections}}:
+                  <ol>
+                    <li>If |collection|["{{HIDCollectionInfo/usagePage}}"] is
+                    equal to |filter|["{{HIDDeviceFilter/usagePage}}"]:
+                      <ol>
+                        <li>If |filter| does not contain
+                        "{{HIDDeviceFilter/usage}}", then return `true`.
```

Comment:
To support the case where a filter contains both vendor and product ID fields as well as usage fields these checks should be rewritten to return only when the filter is sure not to match.

Reply:
Done